src/commands/operator/index.ts

```diff
@@ -208,57 +237,77 @@ export default class Operator extends OperatorJobAwareCommand {
     CliUx.ux.action.stop('🚀')
+  }
 
-    // check if file exists
-    if (await fs.pathExists(this.jobsFile)) {
-      this.log('Saved jobs file exists, parsing it for valid/active jobs.')
-      // if file exists, need to add it to list of jobs to process
-      this.operatorJobs = (await fs.readJson(this.jobsFile)) as {[key: string]: OperatorJob}
-      // need to check each job and make sure it's still valid
-      for (const jobHash of Object.keys(this.operatorJobs)) {
-        this.operatorJobs[jobHash].gasLimit = BigNumber.from(this.operatorJobs[jobHash].gasLimit)
-        this.operatorJobs[jobHash].gasPrice = BigNumber.from(this.operatorJobs[jobHash].gasPrice)
-        this.operatorJobs[jobHash].jobDetails.startTimestamp = BigNumber.from(
-          this.operatorJobs[jobHash].jobDetails.startTimestamp,
-        )
-        // if job is still valid, it will stay in object, otherwise it will be removed
-        // Tags not passed in because they do not exist
-        // Maybe save tags with the job hash so we can pass it back in here
-        await this.checkJobStatus(jobHash)
+  async processSavedJobs(): Promise<void> {
+    try {
+      // Check if file exists
+      if (await fs.pathExists(this.jobsFile)) {
+        this.log('Saved jobs file exists, parsing it for valid/active jobs.')
+        // if file exists, need to add it to list of jobs to process
+        this.operatorJobs = (await fs.readJson(this.jobsFile)) as {[key: string]: OperatorJob}
+
+        // Need to check each job and make sure it's still valid
+        for (const jobHash of Object.keys(this.operatorJobs)) {
+          this.operatorJobs[jobHash].gasLimit = BigNumber.from(this.operatorJobs[jobHash].gasLimit)
+          this.operatorJobs[jobHash].gasPrice = BigNumber.from(this.operatorJobs[jobHash].gasPrice)
+          this.operatorJobs[jobHash].jobDetails.startTimestamp = BigNumber.from(
+            this.operatorJobs[jobHash].jobDetails.startTimestamp,
+          )
+
+          // If job is still valid, it will stay in object, otherwise it will be removed
+          await this.checkJobStatus(jobHash)
+          this.log('Saved jobs parsing completed.')
+        }
+      } else {
+        this.log('Saved jobs file not found (not loaded).')
       }
-    } else {
-      this.log('Saved jobs file not found (not loaded).')
+    } catch (error) {
+      this.handleErrorAndExit('An error occurred while processing saved jobs', error)
     }
+  }
 
+  scheduleJobsProcessing(): void {
     for (const network of this.networkMonitor.networks) {
-      // instantiate all network operator job watchers
-      setTimeout(this.processOperatorJobs.bind(this, network), 60_000)
+      // This starts processing jobs after an initial delay (currently set to 0 seconds)
+      setTimeout(async () => {
+        // eslint-disable-next-line no-constant-condition
+        while (true) {
+          try {
+            await this.processOperatorJobs(network)
```

Comment:
So I like this change to make processing more linear. One trade off is that this may also cause delays if this function takes time to execute. Just calling out the trade off.